File: src/main/java/com/freya02/botcommands/api/annotations/RequireOwner.java
package com.freya02.botcommands.api.annotations;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Use this annotation if you want the command to be usable only by the bot owners
 * <p>
 * <b>This is only usable on text commands</b>
 */
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.TYPE, ElementType.METHOD})
public @interface RequireOwner {}

File: src/main/kotlin/com/freya02/botcommands/api/commands/annotations/RequireOwner.kt
package com.freya02.botcommands.api.commands.annotations

import com.freya02.botcommands.api.commands.prefixed.builder.TextCommandBuilder

/**
 * Marks this text command as only usable by the bot owners.
 *
 * @see TextCommandBuilder.ownerRequired DSL equivalent
 */
@Target(AnnotationTarget.CLASS, AnnotationTarget.FUNCTION)
@Retention(AnnotationRetention.RUNTIME)
annotation class RequireOwner